src/ui/components/confirm.c

```diff
@@ -167,5 +170,6 @@ component_t* confirm_create(
                 _on_confirm));
     }
 
+    screen_saver_disable();
```

Comment:
adding these two functions sounds good, but scattering these in C code like this is a bit fragile:

- confirm_create() might be called without actually being shown, in which case the screensaver would stay disabled
- it is hard to cover all UI workflows like this, e.g.: password input, sdcard prompt, etc.

Also, doing it in C is harder to reason about due to all the callbacks, whereas in Rust, there are async functions where it would be easier.

For example, the confirm workflow in async rust is [here](https://github.com/digitalbitbox/bitbox02-firmware/blob/166deaf431c0d4dd0304593c8a0cefcf038babb6/src/rust/bitbox02-rust/src/workflow/confirm.rs#L34). It seems that many components use the `option(&result).await` function at the end to wait for the component to finish.

Maybe a better solution then is to put the screensaver disable/enable calls inside the `option(...)` future, covering many components at once. Easiest would probably be to add a `async fn option_no_screensaver(...) { ... }` function and use that everywhere instead of `option(...)` where applicable.

Are you comfortable in Rust or want to give it a shot? If not, let me know and I can help.

Reply:
> My question is: are the enable / disable functions of the screensaver still written in C and I would call them from rust? If yes, then I think I understood it, will try it out. :)

Yeah the C functions remain, and can be called from Rust. You may have to add them to CMakeLists.txt allow-list and wrapper.h.

> EDIT: Maybe another question: Why are some workflows in C and in rust then? Would the option call really work for the workflows then?

There are no workflows left in C except for the first orientation screen and the waiting/idle screen. Maybe you meant components? They are still in C (not migrated to Rust yet), but they are all invoked from Rust workflows.